idarling/core/hooks.py

```diff
@@ -525,15 +526,15 @@ def sgr_changed(self, start_ea, end_ea, regnum, value, old_value, tag):
     #     return 0
 
     def make_data(self, ea, flags, tid, size):
-        # TODO: tid --> struct name
         self._plugin.logger.debug("make_data(ea = %x, flags = %x, tid = %x, size = %x)" % (ea, flags, tid, size))
-        self._send_packet(evt.MakeDataEvent(ea, flags, size, tid))
+        self._send_packet(evt.MakeDataEvent(ea, flags, size, ida_struct.get_struc_name(tid) if tid != ida_idaapi.BADADDR else ''))
         return 0
 
     def renamed(self, ea, new_name, local_name):
         old_name = ""
-        if ida_struct.is_member_id(ea):
-            old_name = ida_struct.get_struc_name(ea)
+        if ida_struct.is_member_id(ea) or ida_struct.get_struc_name(ea) or ida_enum.get_enum_name(ea):
+            # old_name = ida_struct.get_struc_name(ea)
+            return 0 #drop hook for avoid  dublicate 'StrucRenamedEvent', 'EnumRenamedEvent' and 'StrucMemberRenamedEvent'
```

Comment:
typo: dublicate => duplicate

Reply:
Yep, apparently, I was thinking about word "doubled" in this moment)). And, idaapi wonder:
```
rodata:08056D08 test_2324       dd 0
```
```
Python>print(ida_struct.get_struc_name(0x8056d08))
test_2324
```
Changing `get_struc_name` to `get_struc`. Looks workable.